compiler/damlc/daml-ide/src/DA/Daml/LanguageServer.hs

```diff
@@ -68,6 +77,39 @@ setIgnoreOptionalHandlers = PartialHandlers $ \WithMessage{..} x -> return x
     -- $/setTraceNotification which we want to ignore.
     where optionalPrefix = "$/"
 
+filesFromExecParams :: List Aeson.Value -> [NormalizedFilePath]
+filesFromExecParams (List files) = map (toNormalizedFilePath . T.unpack) (concatMap fileStringToPath files)
+            where fileStringToPath :: Aeson.Value -> [T.Text]
+                  fileStringToPath (Aeson.String x) = [x]
+                  fileStringToPath (Aeson.Array arr) = concatMap fileStringToPath arr
+                  fileStringToPath _ex = error ("Failed to get daml files from workspace" ++ show _ex )
+
+onCommand
+    :: IdeState
+    -> ExecuteCommandParams
+    -> IO Aeson.Value
+onCommand ide ExecuteCommandParams{..} = do
+    case _arguments of
+        Nothing -> return $ Aeson.String "No .daml files where found in the IDE workspace"
+        Just path -> do
+            case filesFromExecParams path of
+                (head : rest) -> do
+                        logInfo (ideLogger ide) "Generating visualization for current daml project"
+                        mbmodules <- mapM (\f -> runAction ide (useWithStale GenerateDalf f)) rest
+                        Just (WhnfPackage package, _) <- runAction ide (useWithStale GeneratePackage head)
```

Comment:
The pattern match is not safe, see below for an explanation of when this can be `Nothing`.

Reply:
No more doing this